src/Caller.cs

```diff
@@ -21,6 +23,61 @@ internal Caller(IntPtr handle)
             this.store = context.Store;
         }
 
+        /// <summary>
+        /// Gets a Span from an exported memory of the caller by the given name.
+        /// </summary>
+        /// <param name="name">The name of the exported memory.</param>
+        /// <param name="address">The zero-based address of the start of the span.</param>
+        /// <param name="length">The length of the span.</param>
+        /// <param name="result">The Span of memory (if the function returns true)</param>
+        /// <returns>Returns true if the exported memory is found or false if a memory of the requested name is not exported.</returns>
+        /// <remarks>
+        /// <para>
+        /// The span may become invalid if the memory grows.
+        ///
+        /// This may happen if the memory is explicitly requested to grow or
+        /// grows as a result of WebAssembly execution.
+        /// </para>
+        /// <para>
+        /// Therefore, the returned span should not be used after calling the grow method or
+        /// after calling into WebAssembly code.
+        /// </para>
+        /// <para>
+        /// Note that WebAssembly always uses little endian as byte order. On platforms 
+        /// that use big endian, you will need to convert numeric values accordingly.
+        /// </para>
+        /// </remarks>
+        public bool TryGetMemorySpan<T>(string name, long address, int length, out Span<T> result)
+            where T : unmanaged
+        {
+            var nameLength = Encoding.UTF8.GetMaxByteCount(name.Length);
+            var nameBytes = nameLength < StackallocThreshold ? stackalloc byte[nameLength] : new byte[nameLength];
+            nameLength = Encoding.UTF8.GetBytes(name, nameBytes);
+            nameBytes = nameBytes[..nameLength];
```

Comment:
I would probably prefer `GetByteCount` over `GetMaxByteCount` here (like in `Linker.DefineFunction`):
The latter is faster (it doesn't iterate through a string) but just returns `(length + 1) * 3)` for UTF-8 (as a non-surrogate char > U+07FF would need 3 bytes), so in the common case when the string consists just of ASCII characters, the max count would be three times the string length, and so it's more likely to exceed the `StackallocThreshold` (when `name.Length >= 85`).

What do you think?
```suggestion
            var nameLength = Encoding.UTF8.GetByteCount(name);
            var nameBytes = nameLength < StackallocThreshold ? stackalloc byte[nameLength] : new byte[nameLength];
            Encoding.UTF8.GetBytes(name, nameBytes);
```

Reply:
Yeah I wasn't sure what the best tradeoff betwee speed/accuracy would be there. I'm happy to swap to `GetByteCount` instead if you think that's the better balance (random aside: I _really_ wish C# allowed us to write a helper function that stack allocated in the caller context, then we could centralise this kind of decision!).